core/src/main/java/io/micronaut/core/annotation/BuildTimeInit.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotate that stores references to class names that can be build time initialized inside bean definition metadata.
+ *
+ * <p>This annotation is currently designed for internal use only and not for public API.</p>
+ */
+@Experimental
+@UsedByGeneratedCode
+@Internal
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Make it `SOURCE` we don't need this annotation at runtime

Reply:
unfortunately `SOURCE` annotations cannot be read by native image features so don't know how to do this

Reply:
Are you sure? Because that's using our own AnnotationMetadata

Reply:
yes but the runtime metadata

Reply:
if I change it to `SOURCE` the test fails